S/samtools/build_tarballs.jl

```diff
@@ -0,0 +1,49 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "samtools"
+version = v"1.10.0"
```

Comment:
Any particular reasons for using this version and not [v1.14](https://github.com/samtools/samtools/releases/tag/1.14) which is the latest one?

Reply:
I tried to use v1.14 but I hit an error building that seemed related to htslib_jll that I was using (which is at 1.10). I tried downgrading to the same version and that seemed to fix the issue.

I'm happy to try and make a htslib 1.14 in the future and then an equivalent samtools, but I thought I better get my feet wet first :)

Reply:
> I'm happy to try and make a htslib 1.14 in the future 

Too late, I did it a few hours ago: #3879 🙂 what do you want to do? Merge this as is or upgrade to v1.14?

Reply:
Awesome, I'm all for upgrading to v1.14, would that require a new pull request or can we specify the new version here?

Reply:
You can do it here

Reply:
Ok, I did it for you :slightly_smiling_face: